### PR TITLE
Switch linters to golangci-lint

### PR DESCRIPTION
**Pull request by haydnKing**
`gometalinter` is dead. See: https://github.com/alecthomas/gometalinter/issues/590

`golangci-lint` is the new kid on the block. See: https://github.com/golangci/golangci-lint

This PR swaps out one for the other, and fixes a bunch of self-contained linter errors spotted by the new tool. Mostly they relate to test code (which was apparently barely linted previously), but there are some more interesting cases, e.g.
- HTTP requests with arbitrary strings
- not storing the result of type-switches in a variable
- using `strings.ToLower(a) == strings.ToLower(b)` instead of `strings.EqualFold(a, b)`

### CODE REVIEW

File: antha/AnthaStandardLibrary/Packages/enzymes/Ligation.go [hunk 1]
```diff
@@ -644,7 +644,8 @@ func MultipleAssemblies(parameters []Assemblyparameters) (s string, successfulas
 			if strings.Contains(err.Error(), "Failure Joining fragments after digestion") {
 				constructsitesstring := make([]string, 0)
 				constructsitesstring = append(constructsitesstring, output)
-				enzyme, err := lookup.RestrictionEnzyme(construct.Enzymename)
```
Comment (m-i-sadowski) [inline antha/AnthaStandardLibrary/Packages/enzymes/Ligation.go]: I assume this was an issue with variable shadowing?

File: antha/AnthaStandardLibrary/Packages/igem/igem.go [hunk 2]
```diff
@@ -143,39 +143,19 @@ func MakeFastaURL(partname string) (Urlstring string) {
 	return Urlstring
 }
 
-func MakeXMLURL(partnames []string) (Urlstring string) {
```
Comment (m-i-sadowski) [inline antha/AnthaStandardLibrary/Packages/igem/igem.go]: Dead code?

File: antha/AnthaStandardLibrary/Packages/pubchem/pubchem.go [hunk 3]
```diff
@@ -186,21 +186,13 @@ func makeOutputspec(spec string, optionalcallbackname string) (outputspec string
...
-	if operation_options != "" {
-		array = append(array, operation_options)
-	}
-	Urlstring := strings.Join(array, "/")
 	/*<input specification>/<operation specification>/[<output specification>][?<operation_options>]
 
 	http://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/name/glucose/property/MolecularFormula,MolecularWeight/JSON
 	*/
-	res, err := http.Get(Urlstring)
+	res, err := http.Get(fmt.Sprintf("http://pubchem.ncbi.nlm.nih.gov/rest/pug/%s/%s/%s/%s", inputspec, operationspec, outputspec, operation_options))
```
Comment (m-i-sadowski) [inline antha/AnthaStandardLibrary/Packages/pubchem/pubchem.go]: What is its issue here?

File: microArch/driver/liquidhandling/robotinstruction.go [hunk 4]
```diff
@@ -121,54 +121,54 @@ func (name InstructionParameter) String() string {
...
-	VOLUNT                               = "VOLUNT"
-	WELL                                 = "WELL"
-	WELLFROM                             = "WELLFROM"
-	WELLFROMVOLUME                       = "WELLFROMVOLUME"
-	WELLTO                               = "WELLTO"
-	WELLTOVOLUME                         = "WELLTOVOLUME" // WELLTOVOLUME refers to the volume of liquid already present in the well location for which a sample is due to be transferred to.
-	WELLVOLUME                           = "WELLVOLUME"
-	WHAT                                 = "WHAT"
-	WHICH                                = "WHICH" // WHICH returns the Component IDs, i.e. representing the specific instance of an LHComponent not currently implemented.
+	CHANNEL         InstructionParameter = "CHANNEL"
```
Comment (m-i-sadowski) [inline microArch/driver/liquidhandling/robotinstruction.go]: I'm not convinced this is an improvement

File: antha/AnthaStandardLibrary/Packages/enzymes/Ligation.go (hunk 1 above)
Comment (haydnKing) [reply antha/AnthaStandardLibrary/Packages/enzymes/Ligation.go]: yeah, later on in this function there was a shadowed err that was never checked

File: antha/AnthaStandardLibrary/Packages/pubchem/pubchem.go (hunk 3 above)
Comment (haydnKing) [reply antha/AnthaStandardLibrary/Packages/pubchem/pubchem.go]: the whole of the issue here is that `gosec` really doesn't like `http.Get(randomString)`... which is perhaps a bit overkill, but given that this is callable from elements I figured it was a good prompt to tidy up here a bit

File: microArch/driver/liquidhandling/robotinstruction.go (hunk 4 above)
Comment (haydnKing) [reply microArch/driver/liquidhandling/robotinstruction.go]: yeah, me neither to be honest... can look into whether I can specifically disable this option

File: antha/AnthaStandardLibrary/Packages/pubchem/pubchem.go (hunk 3 above)
Comment (m-i-sadowski) [reply antha/AnthaStandardLibrary/Packages/pubchem/pubchem.go]: If it were likely to happen often I'd suggest this is a candidate for switching off. As it is it doesn't seem terribly likely to be a huge issue

File: microArch/driver/liquidhandling/robotinstruction.go (hunk 4 above)
Comment (m-i-sadowski) [reply microArch/driver/liquidhandling/robotinstruction.go]: OK so as you pointed out, https://staticcheck.io/docs/checks#SA9004, proves I'm entirely wrong! 

_slinks off_

**Pull request merged**